StudyHub/StudyHub.BLL/Services/SubjectService.cs

```diff
@@ -26,6 +29,30 @@ public class SubjectService : ISubjectService
         _mapper = mapper;
     }
 
+    public async Task<List<StudentDTO>> AddStudentToSubjectAsync(Guid subjectId, AddStudentSubjectRequest request)
+    {
+        var users = new List<User>();
+
+        foreach (var email in request.Emails)
+        {
+            var user = await _userManager.FindByEmailAsync(email)
+                ?? throw new NotFoundException($"Student not found with the specified email: {email}");
+
+            users.Add(user);
+        }
+
+        var subject = await _subjectRepository.FirstOrDefaultAsync(s => s.Id == subjectId)
+            ?? throw new NotFoundException("Subject not found with the specified Id");
+
+        subject.Students ??= new List<User>();
+
+        subject.Students.AddRange(users);
+
+        await _subjectRepository.SaveChangesAsync();
```

Comment:
there is no need to use SaveChangesAsync. You can add students like 
subject.Students = users
_subjectRepository.UpdateAsync(subject)

Reply:
SaveChangesAsync dont uses in Services, it looks weird

Reply:
Why is it weird? SaveChanges is used when you want to save changes. I added the changes to the StudentSubject table and saved them.

Reply:
And if we use _subject.Students = users_ the new data will not be added to the existing ones but will replace them.

Reply:
SaveChanges its an dal responsibility, not bll, and that`s why I think using it in service unjustified. If you want to update data, u can create some temp value -> copy exiting list in this temp -> add new records -> update entity. On my mind, it will looks better

Reply:
SaveChanges its an dal responsibility, not bll, and that`s why I think using it in service unjustified. If you want to update data, u can create some temp value -> copy exiting list in this temp -> add new records -> update entity. On my mind, it will looks better

Reply:
@aquaprogit what you say about it ?